pe_scope_guard/pe_scope_guard.hpp

```diff
@@ -0,0 +1,31 @@
+#ifndef PE_SCOPE_GUARD_HPP
+#define PE_SCOPE_GUARD_HPP
+
+#include <concepts>
+#include <tuple>
+#include <type_traits>
+#include <utility>
+
+namespace peach
+{
+
+  template< typename... Tys > class ScopeGuard
+  {
+  public:
+    constexpr ScopeGuard( Tys... callables ) requires( ( std::same_as< std::invoke_result_t< Tys >, void > && ... ) )
+        : m_callables( std::make_tuple( callables... ) )
+    {
+    }
+
+    ~ScopeGuard( ) noexcept
+    {
+      std::apply( []( auto&&... args ) { ( ( args( ) ), ... ); }, m_callables );
+    }
```

Comment:
What if one of the callables actually throws in the destructor, right now it would call `std::terminate()` because of the `noexcept`.

What if we `try{} catch(...) {  ... }` instead and throw a `peach::Exception("Function passed to ScopeGuard threw an exception")` or something similar. 

Thoughts?

Reply:
The problem is that if the destructor throws an exception it does not handle, the program will call `std::terminate()` while stack-unwinding. So generally speaking we should not throw any exception in the destructor that we dont handle. 
Also destructors have the same exception specification as the implicitly-defined destructor would have, which in most cases is `noexcept`. So to be able to have a throwing destructor you need to declare it as `noexcept(false)`.

Reply:
Conclusion: Will not throw any custom exception in here for the reasons state above